lib/elixir/lib/calendar/naive_datetime.ex

```diff
@@ -720,23 +721,41 @@ defmodule NaiveDateTime do
       iex> NaiveDateTime.to_iso8601(~N[2000-02-28 23:00:13.001])
       "2000-02-28T23:00:13.001"
 
+      iex> NaiveDateTime.to_iso8601(~N[2000-02-28 23:00:13.001], :extended, 0)
+      "2000-02-28T23:00:13.001Z"
+
+      iex> NaiveDateTime.to_iso8601(~N[2000-02-28 23:00:13.001], :extended, 3600)
+      "2000-02-28T23:00:13.001+01:00"
+
+      iex> NaiveDateTime.to_iso8601(~N[2000-02-28 23:00:13.001], :basic)
+      "20000228T230013.001"
+
       iex> NaiveDateTime.to_iso8601(~N[2000-02-28 23:00:13.001], :basic)
       "20000228T230013.001"
 
+      iex> NaiveDateTime.to_iso8601(~N[2000-02-28 23:00:13.001], :basic, 3600)
+      "20000228T230013.001+0100"
+
   This function can also be used to convert a DateTime to ISO 8601 without
-  the time zone information:
+  the time zone information, or override with another offset:
 
       iex> dt = %DateTime{year: 2000, month: 2, day: 29, zone_abbr: "CET",
       ...>                hour: 23, minute: 0, second: 7, microsecond: {0, 0},
       ...>                utc_offset: 3600, std_offset: 0, time_zone: "Europe/Warsaw"}
       iex> NaiveDateTime.to_iso8601(dt)
       "2000-02-29T23:00:07"
 
+      iex> dt = %DateTime{year: 2000, month: 2, day: 29, zone_abbr: "CET",
+      ...>                hour: 23, minute: 0, second: 7, microsecond: {0, 0},
+      ...>                utc_offset: 3600, std_offset: 0, time_zone: "Europe/Warsaw"}
+      iex> NaiveDateTime.to_iso8601(dt, :extended, 7200)
+      "2000-02-29T23:00:07+02:00"
+
   """
-  @spec to_iso8601(Calendar.naive_datetime(), :basic | :extended) :: String.t()
-  def to_iso8601(naive_datetime, format \\ :extended)
+  @spec to_iso8601(Calendar.naive_datetime(), :basic | :extended, nil | integer()) :: String.t()
+  def to_iso8601(naive_datetime, format \\ :extended, offset \\ nil)
```

Comment:
I think the default offset can be zero.

Reply:
I considered that, but I didn't want to change the default behaviour of printing nothing by default, a `0` offset should print as `Z`.

Reply:
We should not print zero as Z as that requires Etc/UTC, which we can't assume here.

Reply:
Unless we want to show zero as "+00:00" and not offset as nothing. But I don't know if that's a thing. Thoughts?

Reply:
Apparently it is a thing. For example, a -01:00 timezone on summer time. So I think we indeed need to treat `nil` and `0` separately. We just need to make sure to show zero as "+00:00" instead of Z. Perhaps I would even make it two distinct function signatures: `to_iso8601/2` and `to_iso8601/3`.

Reply:
Ok I can definitely seperate it to two seperate functions, and I'm sort of ok with `+00:00` but this reintroduces a disparity, a time parsed from `Z` will be treated as UTC but theres no "one way" of parsing it back to `Z`, e.g. the end user has to care about timezones when they don't have any timezone information.

My aim is to make a reversible set of function calls such that given a string you can get back to that string without involving a timezone lookup... Not being able to set `Z` from input causes a problem for that...